qiskit/circuit/classical/expr/constructors.py

```diff
@@ -63,21 +63,7 @@ def _coerce_lossless(expr: Expr, type: types.Type) -> Expr:
 def lift_legacy_condition(
     condition: tuple[qiskit.circuit.Clbit | qiskit.circuit.ClassicalRegister, int], /
 ) -> Expr:
-    """Lift a legacy two-tuple equality condition into a new-style :class:`Expr`.
-
-    Examples:
-        Taking an old-style conditional instruction and getting an :class:`Expr` from its
-        condition::
-
-            from qiskit.circuit import ClassicalRegister
-            from qiskit.circuit.library import HGate
-            from qiskit.circuit.classical import expr
-
-            cr = ClassicalRegister(2)
-            instr = HGate().c_if(cr, 3)
-
-            lifted = expr.lift_legacy_condition(instr.condition)
-    """
+    """Lift a legacy two-tuple equality condition into a new-style :class:`Expr`."""
```

Comment:
Does keeping `lift_legacy_condition` make sense now that the legacy conditions are gone? is there any other use case for it?

Reply:
We can still have a legacy condition structure. It's basically just a tuple of `(register, val)`, it just won't come from a `.condition` anymore, so we still need this function